libraries/aws/provisioning/include/aws_iot_provisioning.h

```diff
@@ -115,12 +116,59 @@ AwsIotProvisioningError_t AwsIotProvisioning_Init( uint32_t mqttTimeout );
  *   #AWS_IOT_PROVISIONING_INTERNAL_FAILURE, if any there are operation failures internal to the library.
  */
 /* @[declare_provisioning_createkeysandcertificate] */
-AwsIotProvisioningError_t AwsIotProvisioning_CreateKeysAndCertificate( IotMqttConnection_t provisioningConnection,
+AwsIotProvisioningError_t AwsIotProvisioning_CreateKeysAndCertificate( IotMqttConnection_t connection,
                                                                        uint32_t flags,
                                                                        uint32_t timeoutMs,
                                                                        const AwsIotProvisioningCreateKeysAndCertificateCallbackInfo_t * pResponseCallback );
 /* @[declare_provisioning_createkeysandcertificate] */
 
+/**
+ * @brief Requests the AWS IoT Core service for a certificate by sending a Certificate-Signing Request, and
+ * invokes the provided user-defined response handler with the response it receives from the server.
+ *
+ * @note It is advised to use a shared MQTT connection to AWS IoT Core across all API functions.
+ *
+ * @warning This function is NOT thread-safe. Concurrent calls to the library API functions can result in undefined
```

Comment:
I see the other functions also follow this pattern, but it seems weird that the `@note`s and `@warning`s are separated by the `@param` list instead of being all together

Reply:
I had intentionally followed the pattern to keep the information of the API functions separate from its parameter list documentation, as it was more readable to me.

Reply:
OK.